pkg/controller/ads/dns.go

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ads
+
+import (
+	"net"
+	"net/netip"
+	"slices"
+	"time"
+
+	clusterv3 "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	endpointv3 "github.com/envoyproxy/go-control-plane/envoy/config/endpoint/v3"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	core_v2 "kmesh.net/kmesh/api/v2/core"
+	"kmesh.net/kmesh/pkg/dns"
+)
+
+// adsDnsResolver is DNS resolver of Kernel Native
+type AdsDnsResolver struct {
+	Clusters    chan []*clusterv3.Cluster
+	adsCache    *AdsCache
+	dnsResolver *dns.DNSResolver
+}
+
+// pending resolve domain info of Kennel-Native Mode,
+// domain name is used for dns resolution
+// cluster is used for create the apicluster
+type pendingResolveDomain struct {
+	DomainName  string
+	Clusters    []*clusterv3.Cluster
+	RefreshRate time.Duration
+}
+
+func NewAdsDnsResolver(adsCache *AdsCache) (*AdsDnsResolver, error) {
+	resolver, err := dns.NewDNSResolver()
+	if err != nil {
+		return nil, err
+	}
+	return &AdsDnsResolver{
+		Clusters: make(chan []*clusterv3.Cluster),
+		// dnsRefreshQueue: workqueue.NewTypedDelayingQueueWithConfig(workqueue.TypedDelayingQueueConfig[any]{Name: "dnsRefreshQueue"}),
+		adsCache:    adsCache,
+		dnsResolver: resolver,
+	}, nil
+}
+
+func (adsResolver *AdsDnsResolver) StartAdsDnsResolver(stopCh <-chan struct{}) {
+	go adsResolver.startAdsResolver()
+	go adsResolver.dnsResolver.StartDnsResolver()
```

Comment:
I feel this is messy cand confusing, it is unclear what is AdsResolver and what is dns resolver